### PR TITLE
base1: Various cleanups

### PR DESCRIPTION
**Pull request by martinpitt**
This is the first set of cleanups for my eventual goal of eliminating jquery from base1 (the external  API got deprecated years ago), building base1 from webpack, and thus simplifying our Makefiles.

### CODE REVIEW

File: src/base1/test-chan.js [hunk 1]
```diff
@@ -254,11 +255,12 @@ QUnit.test("multiple", function (assert) {
...
     var channelb = cockpit.channel({ host: "amy" });
 
-    $(mock_peer).on("recv", function(event) {
+    const onrecv = event => {
         $(mock_peer).off("recv");
         assert.notStrictEqual(channel.id, channelb.id, "channels have different ids");
         done();
-    });
+    };
+    $(mock_peer).on("recv", onrecv);
```
Comment (martinpitt) [inline src/base1/test-chan.js:263]: FTR, I split out the function as with standard event handlers it needs to have a name for `removeEventHandler()`. That doesn't hurt with jQuery, so it just makes the future porting easier.

**Comment by martinpitt** [comment]
test-locale was reasonably straightforward to port after all, pushed. So now it's down to that single pesky test-chan.html test!

File: src/base1/test-chan.js [hunk 2]
```diff
@@ -337,31 +339,32 @@ QUnit.test("queue messages", function (assert) {
     });
 });
 
-QUnit.test("receive message", function (assert) {
+QUnit.skip("receive message", function (assert) {
```
Comment (marusak) [inline src/base1/test-chan.js:342]: Why do you skip bunch of tests?

Comment (martinpitt) [reply src/base1/test-chan.js:342]: Argh, crap! That was the one I was debugging for ages, and then I forgot to take out the skip again. Thanks for spotting! I'll most probably revert to the "port most tests" commit again, and leave that "receive message" sucker for later (the *only* test that throws a spanner into the works, and I just can't figure out for the life of me why the message isn't being sent).

**Comment by martinpitt** [comment]
I added a little hack so that we can keep dropping the unit tests' dependency on base1/jquery.js. This will help with #14892.

**Comment by mvollmer** [comment]
mustache is still used, at least by `pkg/lib/machine-dialogs.js`.  But we don't load it from base1 anymore. I wouldn't call it "deprecated" and "unused" in general.

**Comment by mvollmer** [comment]
Would be nice to get rid of `cockpit.event_target` as well, somehow....

**Comment by martinpitt** [comment]
> I wouldn't call it "deprecated" and "unused" in general.

No, it's not of course, but the base1/mustache.js file most definitively is. That's what I want to get rid of.

**Comment by mvollmer** [comment]
> > I wouldn't call it "deprecated" and "unused" in general.
> 
> No, it's not of course, but the base1/mustache.js file most definitively is. That's what I want to get rid of.

Yeah. I think just saying "The base1/mustache.js file hasn't been used on any page for years, and has never been
documented as external API." would avoid potential confusing.

**Review by mvollmer — CHANGES_REQUESTED** [review]
Let's change the mustache commit message to: "The base1/mustache.js file hasn't been used on any page for years, and has never been documented as external API."
Otherwise LGTM!

**Comment by martinpitt** [comment]
@mvollmer: Updated (with push-rewrite) to adjust the commit message.

**Review by mvollmer — APPROVED** [review]

**Pull request merged**